### PR TITLE
feat(calendar): add calendar & small refactor

### PR DESCRIPTION
**Pull request by CheeBam**

### CODE REVIEW

File: src/screens/Calendar/templates/DayView.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import { DateTime, Duration, Interval } from 'luxon';
```
Comment (CheeBam) [inline src/screens/Calendar/templates/DayView.tsx:1]: refactor imports

File: src/screens/Calendar/templates/TaskView.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    <View key={task.id} style={[styles.taskView, task.styles]}>
+      {task.showContent && (
+        <View style={styles.taskContent}>
+          <Text>{task.title}</Text>
+          {task.showRightContent && (
+            <View style={styles.taskRightContent}>
+              <TimeComponent timeInSeconds={task.duration} />
+              <Icon name="pencil" color="#979797" onPress={navigateToEdit} />
```
Comment (art1990) [inline src/screens/Calendar/templates/TaskView.tsx:36]: I would put the color in constants

File: src/screens/Calendar/templates/DayView.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+              top: Math.floor(minutesFromMidnightToStart * MINUTE_HEIGHT),
+              left: task.leftCoefficient * width + '%',
+            },
+            showContent: height > 20,
+            showRightContent: width === 100,
+            ...task,
+          };
+        })
+    );
+  }, [tasks]);
```
Comment (art1990) [inline src/screens/Calendar/templates/DayView.tsx:104]: You need two maps??

File: src/screens/Calendar/templates/DayView.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            tempFinishTime = task.completeAt;
+          }
+
+          return {
+            leftCoefficient,
+            separateIndex,
+            ...task,
+          };
+        })
+        .map(task => {
```
Comment (art1990) [inline src/screens/Calendar/templates/DayView.tsx:73]: You need two maps?

**Comment by DimaMakarenko** [comment]
all good

**Pull request merged**